### ISSUE TITLE
Would like a precise way to map PerformanceTimings to FetchEvent

### ISSUE THREAD

**Issue by jakearchibald** [w3c/resource-timing#259]
From https://github.com/w3c/ServiceWorker/issues/1179 by @clayw

ServiceWorkerGlobalScope.performance.getEntries() gives me all of the performance timing data that's gone through the service worker, but there's no exact mapping back to the original FetchEvent or client to which they originated. Multiple FetchEvents can have the same URL and thus the same PerformanceEntry.name field, for example.

Some strategies for dealing with this may be to look at the sequencing of when they come in or mapping the actual timings back to the event. But it seems like it shouldn't be necessary to try and do any complicated guesswork, especially if there may be some edge cases (eg, loading two tabs simultaneously, quirks with navigation preload, etc).

Can there be some way to exactly link these via API? For example:
 - WorkerPerformance.getEntriesByClientId(clientId) call to filter by client ids
 - Adding a clientId field into the PerformanceResourceTiming somehow
 - Some sort of thenable from FetchEvent like FetchEvent.getResourceTimings()

**Comment by annevk** [comment]
Yeah, this seems to be a problem that exists outside service workers as well. If you fetched `/x` multiple times or some such you'd have to keep track of order yourself.

**Comment by monica-ch** [comment]
We are intending to write an explainer to address this problem, and let me know if you have any additional inputs please comment

**Comment by noamr** [comment]
> I think it would help a lot if someone first described the actual model that underpins RT objects. Currently if you read through Fetch it's entirely unclear how RT objects end up getting allocated and where they exist.
> 
> Once that's clear (and someone verified it reasonably maps to implementations) it would be easier to come up with a good API.

That's done now :)

**Comment by noamr** [comment]
If I understand what's proposed in the OP, I think it's a model that breaks the moment you have caches, because the response that's in the cache might have a different `clientId` than the one that triggered the `FetchEvent` (the cache is shared between clients).

This whole thing is a bit tricky because in a good chunk of the interesting cases there is no 1:1 mapping between a `Response` inside the service worker and a response in the client. The only thing that ties them together is `respondWith`, which is where I think an API that addresses this use case should live: something like allowing to annotate the client's resource timing entry (perhaps adding `Server-Timing` or some such?) in the `respondWith` arguments. 

An exception to this is static routing, where the mapping between responses is a lot cleaner.
